fendermint/vm/topdown/proof-service/src/verifier.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022-2025 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! Proof bundle verification for block attestation
+//!
+//! Provides deterministic verification of proof bundles against F3 certificates.
+//! Used by validators during block attestation to verify parent finality proofs.
+
+use crate::assembler::{NEW_POWER_CHANGE_REQUEST_SIGNATURE, NEW_TOPDOWN_MESSAGE_SIGNATURE};
+use anyhow::Result;
+use cid::Cid;
+use filecoin_f3_certs::FinalityCertificate;
+use proofs::proofs::common::bundle::{UnifiedProofBundle, UnifiedVerificationResult};
+use proofs::proofs::events::bundle::EventProofBundle;
+use proofs::proofs::events::verifier::verify_event_proof;
+use proofs::proofs::storage::verifier::verify_storage_proof;
+
+use proofs::proofs::common::evm::{ascii_to_bytes32, extract_evm_log, hash_event_signature};
+
+pub struct ProofsVerifier {
+    events: Vec<Vec<[u8; 32]>>,
+}
+
+impl ProofsVerifier {
+    pub fn new(subnet_id: String) -> Self {
+        let events = vec![
+            vec![
+                hash_event_signature(NEW_TOPDOWN_MESSAGE_SIGNATURE),
+                ascii_to_bytes32(&subnet_id),
+            ],
+            vec![hash_event_signature(NEW_POWER_CHANGE_REQUEST_SIGNATURE)],
+        ];
```

Comment:
Is this to limit what kind of events are allowed in proof bundles?

Reply:
Yes, exactly.